test/test_nestedtensor.py

```diff
@@ -4347,6 +4361,138 @@ def forward(self, query, value, offsets):
+    @unittest.skipIf(IS_WINDOWS, reason="Windows not yet supported for torch.compile")
+    @skipCUDAIf(not SM70OrLater, "GPU capability is < SM70")
+    def test_compile_with_dynamic_min_seq_len(self, device, dtype):
+        # shape (B, *, D)
+        # min seq len: 7
+        nt = torch.nested.nested_tensor([
+            torch.randn(7, 5),
+            torch.randn(8, 5),
+            torch.randn(9, 5),
+        ], layout=torch.jagged)
+
+        # min seq len: 8
+        nt2 = torch.nested.nested_tensor([
+            torch.randn(8, 5),
+            torch.randn(9, 5),
+            torch.randn(10, 5),
+        ], layout=torch.jagged)
+
+        def f(nt):
+            return torch.ones_like(nt) * nt.min_seqlen()
+
+        for dynamic in [False, True, None]:
+            self.assertFalse(_recompiles_for_inputs(f, (nt,), (nt2,), dynamic=dynamic))
+
+    @dtypes(torch.float32)
+    @skipIfTorchDynamo("Test compiles internally")
+    @unittest.skipIf(sys.version_info >= (3, 12), "torch.compile is not supported on python 3.12+")
+    @unittest.skipIf(IS_WINDOWS, reason="Windows not yet supported for torch.compile")
+    @skipCUDAIf(not SM70OrLater, "GPU capability is < SM70")
+    def test_compile_with_propagated_dynamic_max_seq_len(self, device, dtype):
+        # shape (B, *, D)
+        # max seq len: 18
+        nt = torch.nested.nested_tensor([
+            torch.randn(2, 5),
+            torch.randn(3, 5),
+            torch.randn(18, 5),
+        ], layout=torch.jagged)
+
+        # max seq len: 19
+        nt2 = torch.nested.nested_tensor([
+            torch.randn(2, 5),
+            torch.randn(3, 5),
+            torch.randn(19, 5),
+        ], layout=torch.jagged)
+
+        def f(nt):
+            nt2 = nt.sin() + 1
+            return torch.ones_like(nt2) * nt2.max_seqlen()
+
+        ref = f(nt)
+        output = torch.compile(f, fullgraph=True, dynamic=False)(nt)
+        self.assertEqual(ref, output)
+
+        for dynamic in [False, True, None]:
+            self.assertFalse(_recompiles_for_inputs(f, (nt,), (nt2,), dynamic=dynamic))
+
+    @dtypes(torch.float32)
+    @skipIfTorchDynamo("Test compiles internally")
+    @unittest.skipIf(sys.version_info >= (3, 12), "torch.compile is not supported on python 3.12+")
+    @unittest.skipIf(IS_WINDOWS, reason="Windows not yet supported for torch.compile")
+    @skipCUDAIf(not SM70OrLater, "GPU capability is < SM70")
+    def test_compile_with_manually_set_seq_lens(self, device, dtype):
+        # shape (B, *, D)
+        # max seq len: 10
+        nt = torch.nested.nested_tensor([
+            torch.randn(2, 5),
+            torch.randn(3, 5),
+            torch.randn(10, 5),
+        ], layout=torch.jagged)
+
+        # max seq len: 11
+        nt2 = torch.nested.nested_tensor([
+            torch.randn(2, 5),
+            torch.randn(3, 5),
+            torch.randn(11, 5),
+        ], layout=torch.jagged)
+
+        # TODO: demonstrate usage of new API with manually set seq lengths
```

Comment:
Is this TODO planned for this PR?

Reply:
Yeah, I'm working on this today :) might boot it still, TBD

Reply:
Okay, I think I'm going to boot it. In my head, the ideal state is to have public `@property` accessors for both getting and setting min / max sequence length. There's some work being done right now for better `@property` tracing, so I think we should wait on that.